File: qualifier/webapp/java-avans/src/main/java/net/isucon/isucon4/repository/LoggingRepository.java
/*
 * The MIT License (MIT)
 *
 * Copyright (c) 2015 Manabu Matsuzaki
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

package net.isucon.isucon4.repository;

import me.geso.tinyorm.TinyORM;
import net.isucon.isucon4.row.User;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.Arrays;
import java.util.Date;

@Singleton
public class LoggingRepository {

    @Inject
    private TinyORM orm;

    public void create(boolean succeeded, String login, String ip, User user) {
        Integer userId = user != null ? user.getId() : null;
        orm.updateBySQL(
                "INSERT INTO login_log(created_at, user_id, login, ip, succeeded)" +
                        " values (?, ?, ?, ?, ?)",
                Arrays.asList(new Date(), userId, login, ip, succeeded));
    }
}

File: qualifier/webapp/scala-play-scalikejdbc/app/repositories/LoggingRepository.scala
package repositories

import java.time.LocalDateTime
import javax.inject.Singleton

import entities.{LoginLog, Users}
import scalikejdbc._

@Singleton
class LoggingRepository {

  def create(succeeded: Boolean, login: String, ip: String, user: Users)(implicit session: DBSession = AutoSession): Int = {
    val userId = if (user != null) {
      Some(user.id)
    } else {
      None
    }

    val l = LoginLog.column
    applyUpdate {
      insert.into(LoginLog).namedValues(
        l.createdAt -> LocalDateTime.now(),
        l.userId -> userId,
        l.login -> login,
        l.ip -> ip,
        l.succeeded -> succeeded
      )
    }
  }
}
